src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -570,7 +570,7 @@ public static object DynamicInvokeParamHelperCore(ref ArgSetupState argSetupStat
             if (index >= parametersLength)
                 throw new System.Reflection.TargetParameterCountException();
 
-            object incomingParam = argSetupState.parameters[index];
+            object? incomingParam = argSetupState.parameters![index];
```

Comment:
```suggestion
            Debug.Assert(argSetupState.parameters != null);

            object? incomingParam = argSetupState.parameters[index];
```
and also remove `!` from the rest of the `parameters` occurrences below? (I am not 100% sure whether it is going to work.)

Reply:
Will try. That's worst change which I have to do to make it work.